include/seqan3/io/sequence/sequence_file_format_concept.hpp

```diff
@@ -0,0 +1,187 @@
+ * \brief Read from the specified stream and back-insert into the given field buffers.
+ * \memberof seqan3::sequence_file_format_concept
+ * \tparam stream_type      Input stream, must accept any seqan3::istream_concept with `char`.
+ * \tparam options_type     Type of the options, must accept any sequence_file_in::options_type.
+ * \tparam seq_type         Type of the seqan3::field::SEQ input; must accept any seqan3::output_range_concept
+ * over a seqan3::alphabet_concept.
+ * \tparam id_type          Type of the seqan3::field::ID input; must accept any seqan3::output_range_concept
+ * over a seqan3::alphabet_concept.
+ * \tparam qual_type        Type of the seqan3::field::QUAL input; must accept any seqan3::output_range_concept
+ * over a seqan3::quality_concept.
+ * \tparam seq_qual_type    Type of the seqan3::field::SEQ_QUAL input; must accept any seqan3::output_range_concept
+ * over a seqan3::quality_composition.
+ * \param[in,out] stream    The input stream to read from.
+ * \param[in]     opt       File specific options passed to the format.
+ * \param[out]    sequence  The buffer for seqan3::field::SEQ input, i.e. the "sequence".
+ * \param[out]    id        The buffer for seqan3::field::ID input, e.g. the header line in FastA.
+ * \param[out]    quality   The buffer for seqan3::field::QUAL input.
+ * \param[out]    seq_qual  The buffer for seqan3::field::SEQ_QUAL input.
+ *
+ * \details
+ *
+ * ### Additional requirements
+ *
+ *   * The function must also accept std::ignore as parameter for any of the fields. [this is enforced by the concept checker!]
+ *   * In this case the data read for that field shall be discarded by the format.
+ *   * `seq_qual` must be set to std::ignore if either `seq` or `qual` are not set to std::ignore. [this shall be checked by a `static_assert` inside the function]
+ *
+ */
+/*!\fn void write(stream_type & stream, options_type const & opt, seq_type && sequence, id_type && id, qual_type && quality, seq_qual_type && seq_qual)
+ * \brief Write the given fields to the specified stream.
+ * \memberof seqan3::sequence_file_format_concept
+ * \tparam stream_type      Input stream, must accept any seqan3::istream_concept with `char`.
+ * \tparam options_type     Type of the options, must accept any sequence_file_out::options_type.
+ * \tparam seq_type         Type of the seqan3::field::SEQ input; must accept any seqan3::input_range_concept
+ * over a seqan3::alphabet_concept.
+ * \tparam id_type          Type of the seqan3::field::ID input; must accept any seqan3::input_range_concept
+ * over a seqan3::alphabet_concept.
+ * \tparam qual_type        Type of the seqan3::field::QUAL input; must accept any seqan3::input_range_concept
+ * over a seqan3::quality_concept.
+ * \tparam seq_qual_type    Type of the seqan3::field::SEQ_QUAL input; must accept any seqan3::input_range_concept
```

Comment:
hm, i like the `any`s :thinking:

Reply:
Hm, I felt like this was a different scenario than the other ones, because we are not describing the requirement of a function parameter, but we are instead defining a interface that a type is required to implement; it's kind of a double requirement, aimed not at the consumer of an interface, but at the developer of one.

Do you know what I mean?

But I can also change it, it's not really that important.